src/coreclr/tools/superpmi/superpmi-shared/methodcontext.cpp

```diff
@@ -3657,6 +3657,48 @@ void MethodContext::repGetThreadLocalStaticBlocksInfo(CORINFO_THREAD_STATIC_BLOC
     pInfo->offsetOfGCDataPointer                = value.offsetOfGCDataPointer;
 }
 
+void MethodContext::recGetThreadLocalStaticInfo_NativeAOT(CORINFO_THREAD_STATIC_INFO_NATIVEAOT* pInfo)
+{
+    if (GetThreadLocalStaticInfo_NativeAOT == nullptr)
+        GetThreadLocalStaticInfo_NativeAOT = new LightWeightMap<DWORDLONG, Agnostic_GetThreadStaticInfo_NativeAOT>();
+
+    Agnostic_GetThreadStaticInfo_NativeAOT value;
+    ZeroMemory(&value, sizeof(value));
+    value.tlsRootObject       = SpmiRecordsHelper::StoreAgnostic_CORINFO_CONST_LOOKUP(&pInfo->tlsRootObject);
+    value.tlsIndexObject      = SpmiRecordsHelper::StoreAgnostic_CORINFO_CONST_LOOKUP(&pInfo->tlsIndexObject);
+    value.offsetOfThreadLocalStoragePointer       = pInfo->offsetOfThreadLocalStoragePointer;
+    value.threadStaticBaseSlow       = SpmiRecordsHelper::StoreAgnostic_CORINFO_CONST_LOOKUP(&pInfo->threadStaticBaseSlow);
+
+    DWORDLONG key = 1;
+
+    GetThreadLocalStaticInfo_NativeAOT->Add(key, value);
+    DEBUG_REC(dmpGetThreadLocalStaticInfo_NativeAOT(key, result));
+}
+void MethodContext::dmpGetThreadLocalStaticInfo_NativeAOT(DWORDLONG                                     key,
+                                                          const Agnostic_GetThreadStaticInfo_NativeAOT& value)
+{
+    printf("GetThreadLocalStaticInfo_NativeAOT key %016" PRIX64 ", tlsRootObject-%s, tlsIndexObject-%s,  offsetOfThreadLocalStoragePointer-%u, "
+           "threadStaticBaseSlow-%s",
+           key, SpmiDumpHelper::DumpAgnostic_CORINFO_CONST_LOOKUP(value.tlsRootObject).c_str(),
+           SpmiDumpHelper::DumpAgnostic_CORINFO_CONST_LOOKUP(value.tlsIndexObject).c_str(),
+           value.offsetOfThreadLocalStoragePointer,
+           SpmiDumpHelper::DumpAgnostic_CORINFO_CONST_LOOKUP(value.threadStaticBaseSlow).c_str());
+}
+
+void MethodContext::repGetThreadLocalStaticInfo_NativeAOT(CORINFO_THREAD_STATIC_INFO_NATIVEAOT* pInfo)
+{
+    DWORDLONG                              key = 1;
+    Agnostic_GetThreadStaticInfo_NativeAOT value =
+        LookupByKeyOrMiss(GetThreadLocalStaticInfo_NativeAOT, key, ": key %016" PRIX64 "", key);
+
+    DEBUG_REP(dmpGetThreadLocalStaticInfo_NativeAOT(key, value));
+
+    pInfo->tlsRootObject                     = SpmiRecordsHelper::RestoreCORINFO_CONST_LOOKUP(value.tlsRootObject);
+    pInfo->tlsIndexObject                    = SpmiRecordsHelper::RestoreCORINFO_CONST_LOOKUP(value.tlsIndexObject);
+    pInfo->offsetOfThreadLocalStoragePointer = value.offsetOfThreadLocalStoragePointer;
+    pInfo->threadStaticBaseSlow              = SpmiRecordsHelper::RestoreCORINFO_CONST_LOOKUP(value.threadStaticBaseSlow);
+}
+
```

Comment:
Please make sure you test a spmi collection + replay with NativeAOT, so this code is tested.

Reply:
seems to work

<details>
<summary>Logs</summary>

```cmd
D:\git\runtime>croot superpmi.exe -c 652 %CORE_ROOT%\clrjit.dll D:\git\runtime\windows.x64.Checked.mch
Using jit(D:\git\runtime\artifacts\tests\coreclr\windows.x64.Checked\tests\Core_Root\clrjit.dll) with input (D:\git\runtime\windows.x64.Checked.mch)
 indexCount=1 (652)
Jit startup took 6.308800ms
; Assembly listing for method Program:Test1() (FullOpts)
; Emitting BLENDED_CODE for generic X64 - Windows
; FullOpts code
; NativeAOT compilation
; optimized code
; rsp based frame
; partially interruptible
; Final local variable assignments
;
;  V00 OutArgs      [V00    ] (  1,  1   )  struct (32) [rsp+0x00]  do-not-enreg[XS] addr-exposed "OutgoingArgSpace"
;  V01 rat0         [V01,T00] (  8, 10   )    long  ->  rbx         "Final offset"
;  V02 rat1         [V02,T01] (  5,  4   )    long  ->  rdx         "TlsRootAddr access"
;
; Lcl frame size = 32

G_M41298_IG01:  ;; offset=0x0000
       push     rbx
       sub      rsp, 32
                                                ;; size=5 bbWeight=1 PerfScore 1.25
G_M41298_IG02:  ;; offset=0x0005
       lea      rcx, [(reloc 0x4000000000422668)]
       cmp      qword ptr [rcx-0x08], 0
       jne      SHORT G_M41298_IG05
       mov      rcx, qword ptr GS:[0x0058]
       mov      eax, dword ptr [(reloc 0x4000000000422678)]
       mov      edx, (reloc 0x4000000000422680)
       add      rdx, qword ptr [rcx+8*rax]
       mov      rbx, qword ptr [rdx]
       test     rbx, rbx
       je       SHORT G_M41298_IG06
                                                ;; size=46 bbWeight=1 PerfScore 14.00
G_M41298_IG03:  ;; offset=0x0033
       mov      ecx, dword ptr [rbx+0x60]
       call     <unknown method>
       nop
                                                ;; size=9 bbWeight=1 PerfScore 3.25
G_M41298_IG04:  ;; offset=0x003C
       add      rsp, 32
       pop      rbx
       ret
                                                ;; size=6 bbWeight=1 PerfScore 1.75
G_M41298_IG05:  ;; offset=0x0042
       call     CORINFO_HELP_READYTORUN_GCSTATIC_BASE
       mov      rcx, qword ptr GS:[0x0058]
       mov      eax, dword ptr [(reloc 0x4000000000422678)]
       mov      edx, (reloc 0x4000000000422680)
       add      rdx, qword ptr [rcx+8*rax]
       mov      rbx, qword ptr [rdx]
       test     rbx, rbx
       jne      SHORT G_M41298_IG03
                                                ;; size=37 bbWeight=0 PerfScore 0.00
G_M41298_IG06:  ;; offset=0x0067
       mov      rcx, rdx
       lea      rax, [(reloc 0x4000000000422688)]
       call     rax
       mov      rbx, rax
       jmp      SHORT G_M41298_IG03
                                                ;; size=17 bbWeight=0 PerfScore 0.00

; Total bytes of code 120, prolog size 5, PerfScore 32.25, instruction count 31, allocated bytes for code 120 (MethodHash=14305ead) for method Program:Test1() (FullOpts)
; ============================================================

   1: JIT compiled Program:Test1() [FullOpts, IL size=11, code size=120, hash=0x14305ead]
Loaded 1  Jitted 1  FailedCompile 0 Excluded 0 Missing 0
```

</details>